src/internal_events/http_client.rs

```diff
@@ -109,3 +109,17 @@ impl<'a, B: HttpBody> std::fmt::Display for FormatBody<'a, B> {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct HttpBodyDataReceived<'a> {
+    pub buf: &'a Vec<u8>,
+}
+
+impl<'a> InternalEvent for HttpBodyDataReceived<'a> {
```

Comment:
I added lifetimes as the rust compiler was not happy after making the change from data_buf.take().unwrap() to solely using `buf` I believe this should satisfy our use case but I could be wrong. Looking forward to your feedback @bruceg